spec/unit/generator/generator_tags_spec.rb

```diff
@@ -64,7 +64,7 @@ def tags_mash_from_strings(tags_strings)
     context "with excluded and between tags" do
       let(:generator) { GitHubChangelogGenerator::Generator.new(between_tags: %w(1 2 3), exclude_tags: %w(2)) }
       it { is_expected.to be_a Array }
-      it { is_expected.to match_array(tags_mash_from_strings(%w(1 3))) }
+      it { is_expected.to match_array(tags_mash_from_strings(%w(1 2 3))) }
```

Comment:
After thinking a lot about of these changes, I like them.

The new thing is that **excluding tags is a top-level concern**. It's applied as a part of fetching the tags, stripping those excluded away at that point. The semantics on this lower level has changed. For the better – "confident code", as it were.

Here is a test "the old behavior" with this PR's code, which would take exclusions into account on each level:

```ruby
  describe "#get_filtered_tags" do
    subject do
      generator.filter_excluded_tags(
        generator.get_filtered_tags(tags_from_strings(%w(1 2 3 4 5)))
      )
    end

    context "with excluded and between tags" do
      let(:generator) { GitHubChangelogGenerator::Generator.new(between_tags: %w(1 2 3), exclude_tags: %w(2)) }
      it { is_expected.to be_a Array }
      it { is_expected.to match_array(tags_from_strings(%w(1 3))) }
    end
  end

```

The new semantics work like awaage noted:

```ruby
  describe "#get_filtered_tags" do
    subject do
      generator.get_filtered_tags(tags_from_strings(%w(1 2 3 4 5)))
    end

    context "respects between tags" do
      let(:generator) { GitHubChangelogGenerator::Generator.new(between_tags: %w(1 2 3)) }

      it { is_expected.to be_a Array }
      it { is_expected.to match_array(tags_from_strings(%w(1 2 3))) }
    end
  end
```